### PR TITLE
add redirects to who-we-are and what-you-can-do

### LINKED ISSUE #5144: Re-directs part 1
Please redirect a few main nav pages. Redirects should include all languages and all sub-pages.

- [/about](https://foundation.mozilla.org/about/) to [/who-we-are](https://foundation.mozilla.org/who-we-are/) (and it's sub-pages)
- [/participate](https://foundation.mozilla.org/participate/) to [/what-you-can-do](https://foundation.mozilla.org/what-you-can-do/)

Note: primary pages in no language and /en have been done in wagtail redirects but did work for sub-pages and other languages.

### PR DESCRIPTION
**Pull request by cadecairos**
Closes #5144 

Add two localized redirects to networkapi/urls.py:

`/[locale/]about/` -> `/[locale/]who-we-are`
`[locale/]participate` -> `/[locale/]what-we-do`

### CODE REVIEW

**Review by KalobTaulien — APPROVED** [review]
Tested, works as expected. 👍

**Review by Pomax — CHANGES_REQUESTED** [review]
Is there a way for us to do this in a similar way to i18n_patterns? E.g. `urlpatterns = moforedirected(urlpatterns )` before `urlpatterns 
 += i18n_patterns(...)`, so we're not spreading the locations over multiple files?

**Comment by cadecairos** [comment]
> Is there a way for us to do this in a similar way to i18n_patterns? E.g. `urlpatterns = moforedirection(urlpatterns)` before `urlpatterns += i18n_patterns(...)`, so we're not spreading the locations over multiple files?

The URL patterns are wrapped in `i18n_patterns(...)` already, so that the route handlers are applied to all locales handled by Django. Can you elaborate some more on what you envision for this?

**Comment by Pomax** [comment]
I mean "let's make this something we have in a single file, and that can be added in urls.py as one-liner", e.g.:

```
...
from .redirects import mofo_redirects
...

# Custom routes
patterns = [
    # Buyer's Guide / Privacy Not Included
    url(r'^privacynotincluded/', include('networkapi.buyersguide.urls')),

    # Blog RSS feed
    path('blog/rss/', RSSFeed()),
    path('blog/atom/', AtomFeed())
]

# top-level redirects
pattens += mofo_redirects

# wagtail-managed data
patterns.append(url(r'', include(wagtail_urls)))

# Anything that needs to respect the localised
# url format with /<language_code>/ infixed needs
# to be wrapped by django's i18n_patterns feature:
urlpatterns += i18n_patterns(patterns)

...
```

So that we can update redirects in the one file that is intuitively the file where redirects are housed.

File: network-api/networkapi/wagtailpages/views.py [hunk 1]
```diff
@@ -34,4 +34,16 @@ def redirect_to_initiatives(request, subpath):
     if request.META['QUERY_STRING']:
         query_string = f'?{request.META["QUERY_STRING"]}'
 
-    return redirect(f'/{request.LANGUAGE_CODE}/initiatives/{subpath}{query_string}')
+    return redirect(f'/{request.LANGUAGE_CODE}/{destination_path}/{subpath}{query_string}')
+
+
+def redirect_to_whoweare(request, subpath):
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/views.py:40]: It feels like this code should just be directly used in the redirect pattern file, e.g.
```
 url(r'^about/(?P<subpath>.*)', lambda req, path: localized_redirect(req, path, 'who-we-are')),
```

File: network-api/networkapi/urls.py [hunk 2]
```diff
@@ -73,9 +73,8 @@
     path('blog/rss/', RSSFeed()),
     path('blog/atom/', AtomFeed()),
 
-    # redirect /opportunity Wagtail pages to /initiatives
-    # see https://github.com/mozilla/foundation.mozilla.org/issues/2971 for context
-    url(r'^opportunity/(?P<subpath>.*)', redirect_to_initiatives),
+    # Redirects
+    *foundation_redirects(),
```
Comment (Pomax) [inline network-api/networkapi/urls.py:77]: sneeeeeaky

File: network-api/networkapi/wagtailpages/views.py [hunk 3]
```diff
@@ -34,4 +34,4 @@ def redirect_to_initiatives(request, subpath):
     if request.META['QUERY_STRING']:
         query_string = f'?{request.META["QUERY_STRING"]}'
 
-    return redirect(f'/{request.LANGUAGE_CODE}/initiatives/{subpath}{query_string}')
+    return redirect(f'/{request.LANGUAGE_CODE}/{destination_path}/{subpath}{query_string}')
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/views.py:37]: 👍

**Review by Pomax — APPROVED** [review]

**Pull request merged**